clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5962,6 +5962,10 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   if (Args.hasFlag(options::OPT_fsycl, options::OPT_fno_sycl, false))
     Args.AddLastArg(CmdArgs, options::OPT_sycl_std_EQ);
 
+  // Forward -fsycl-default-sub-group-size if in SYCL mode.
+  if (Args.hasFlag(options::OPT_fsycl, options::OPT_fno_sycl, false))
```

Comment:
Apparently -fsycl is not being passed along for the device compile, so this usage is not effective.  I would suggest using `IsSYCLOffloadDevice` to add only for device.  (around line 6683)